test/Quickstart.kt

```diff
@@ -0,0 +1,198 @@
+    @Test
+    fun `Quickstart`() {
+        val schemaString = fileNameToString("./test/data/schema_string.tql")
+        val dataString = fileNameToString("./test/data/data_string.tql")
+        val queryString = fileNameToString("./test/data/query_string.tql")
+
+        runComposeRule(composeRule) {
+            setContent {
+                Studio.MainWindowContent(WindowContext(1000, 500, 0, 0))
+            }
+
+            composeRule.waitForIdle()
+
+            // This opens a dialog box (which we can't see through) so we assert that buttons with that text can be
+            // clicked.
+            composeRule.onAllNodesWithText("Connect to TypeDB").assertAll(hasClickAction())
+
+            StudioState.client.tryConnectToTypeDB(DB_ADDRESS) {}
+            // We wait to connect to TypeDB. This can be slow by default on macOS, so we wait a while.
+            delay(2_500)
+            composeRule.waitForIdle()
+            assertTrue(StudioState.client.isConnected)
+            composeRule.waitForIdle()
+            composeRule.onNodeWithText(DB_ADDRESS).assertExists()
+
+            // Same as connecting to typedb, but we can't see dropdowns either.
+            composeRule.onAllNodesWithText("Select Database").assertAll(hasClickAction())
+
+            try {
+                StudioState.client.tryDeleteDatabase(DB_NAME)
+            }
+            catch (_: Exception) {}
+
+            delay(1_000)
+
+            StudioState.client.tryCreateDatabase(DB_NAME) {}
+            // We wait to create the github database.
+            delay(1_000)
+
+            StudioState.client.tryOpenSession(DB_NAME)
+            // We wait to open the session.
+            delay(1_000)
+
+
+            // This doesn't work because runQuery doesn't source transaction/session types from the GUI.
+            // We could click the relevant buttons as assertions they exist and are clickable and then do a pure-state
+            // query run.
+
+            // Could probably also store the file locally, include it in the test and open the file through the
+            // project browser then use the GUI to operate.
+
+            StudioState.project.tryOpenProject(File("./test/data").toPath())
+            StudioState.appData.project.path = File("./test/data").toPath()
+            composeRule.waitForIdle()
+
+            // Attempting to click these throws an errors since we use a pointer system that requires existence of a
+            // window/awt backed API, but we can't use windows/awt because of limitations in the testing framework.
+
+            // But we can assert that they exist, which is a test unto itself.
+            composeRule.onNodeWithText("schema_string.tql").assertExists()
+            composeRule.onNodeWithText("data_string.tql").assertExists()
+
+            // Check why we aren't updating transaction/session type on click.
+            composeRule.onNodeWithText("schema").performClick()
```

Comment:
Is this comment essentially a `TODO` to act on? If so, the comment should start with "TODO".

Also, how deeply have we investigated this? It's very strange that we need to explicitly open a session and transaction after clicking these buttons.

Reply:
We're going to investigate in the Text Editor tests why this is. This comment has been removed for now as it is no longer relevant.